src/pproc/ecpoint.py

```diff
@@ -0,0 +1,469 @@
+                "thr_inf": np.reshape(thr_inf[index:end_index], (wt_size, num_pred, 1)),
+                "thr_sup": np.reshape(thr_sup[index:end_index], (wt_size, num_pred, 1)),
+            },
+        )
+        temp_wts = np.where(np.any(np.isnan(predictors), axis=0), np.nan, temp_wts)
+
+        wt_allwt += np.einsum("i,i...", codes_wt[index:end_index], temp_wts)
+        wt_rain = numexpr.evaluate(
+            "pred * wts",
+            local_dict={
+                "pred": np.reshape(predictand, (1, num_gp)),
+                "wts": temp_wts,
+            },
+        )
+        cdf_wt = numexpr.evaluate(
+            "sum(wt_rain * (fer + 1), axis=0)",
+            local_dict={
+                "fer": np.reshape(fer[index:end_index], (wt_size, num_fer, 1)),
+                "wt_rain": np.reshape(wt_rain, (wt_size, 1, num_gp)),
+            },
+        )
+        numexpr.evaluate(
+            "pt + cdf", local_dict={"pt": pt_bc_allwt, "cdf": cdf_wt}, out=pt_bc_allwt
+        )
+    return pt_bc_allwt, wt_allwt
+
+
+def compute_weather_types(
+    tp: np.ndarray,
+    cpr: np.ndarray,
+    ws700: np.ndarray,
+    maxmucape: np.ndarray,
+    sr: np.ndarray,
+    bp_loc: str,
+    fer_loc: str,
+    min_predictand: float = 0.04,
+    wt_batch_size: int = 1,
+    ens_batch_size: int = 1,
+) -> Tuple[list[np.ndarray], list[np.ndarray], list[np.ndarray]]:
+    # Extract variables from files
+    bp_file = pd.read_csv(bp_loc, header=0, delimiter=",")
+    fer_file = pd.read_csv(fer_loc, header=0, delimiter=",")
+    bp = bp_file.iloc[:, 1:].to_numpy()
+    fer = fer_file.iloc[:, 1:].to_numpy()
+    codes_wt = bp_file.iloc[:, 0].to_numpy()
+    thr_inf = bp[:, 0:-1:2]
+    thr_sup = bp[:, 1::2]
+
+    predictand = np.where(tp < min_predictand, 0, tp)
+    predictors = np.asarray([cpr, tp, ws700, maxmucape, sr])
+    ens_partial = functools.partial(
+        compute_single_ens,
+        thr_inf=thr_inf,
+        thr_sup=thr_sup,
+        codes_wt=codes_wt,
+        fer=fer,
+        wt_batch_size=wt_batch_size,
+    )
+
+    # inizialize field for the new post-processed ensemble (CDF)
+    # built from all raw ensemble members and all WTs
+    pt_bc_allens_allwt = []
+    # inizialize field for the bias corrected (bc) at grid-scale
+    # fields for all raw ensemble members and all WTs
+    grid_bc_allens_allwt = []
+    # inizialize field for the wt for all raw ensemble members and all WTs
+    wt_allens_allwt = []
+
+    with fut.ProcessPoolExecutor(
+        max_workers=ens_batch_size,
+        initializer=signal.signal,
+        initargs=(signal.SIGTERM, signal.SIG_DFL),
+    ) as executor:
+        for ind_em, result in enumerate(
+            executor.map(
+                ens_partial,
+                *zip(
+                    *[
+                        (predictand[index], predictors[:, index])
+                        for index in range(len(tp))
+                    ]
+                ),
+            )
```

Comment:
Not sure I get this logic. Isn't that equivalent to `executor.map(ens_partial, predictand, predictors.T)`?